test/EndToEndTests/Tests/Client/Build.Desktop/ClientTests/ClientQueryTests.cs

```diff
@@ -176,7 +177,10 @@ public void QueryEntityNavigationWithImplicitKeys()
                     Assert.AreEqual(400, ex.Response.StatusCode);
                     StringResourceUtil.VerifyDataServicesString(ClientExceptionUtil.ExtractServerErrorMessage(ex), "BadRequest_KeyCountMismatch", "Microsoft.Test.OData.Services.AstoriaDefaultService.Message");
                     //InnerException for DataServiceClientException must be set with the exception response frpom the server.
-                    Assert.IsTrue(ex.InnerException.InnerException != null, "InnerException for DataServiceClientException has not been set.");
+                    ODataErrorException oDataErrorException = ex.InnerException.InnerException as ODataErrorException;
+                    Assert.IsTrue(oDataErrorException != null, "InnerException for DataServiceClientException has not been set.");

```

Comment:
Can you used `Assert.ThrowsException` instead?

Reply:
At this point exception is already thrown. What we are checking is the composition of the `Exception `object.

Reply:
Check how `Assert.ThrowsException` works

Reply:
I have checked. But I don't think there is any added benefit to restructure the initial design of the whole `try/catch` block.